parsers/dataField/parseSingleESDTTransfer.go

```diff
@@ -11,13 +11,14 @@ func (odp *operationDataFieldParser) parseSingleESDTTransfer(args [][]byte, func
 		return responseParse
 	}
 
-	if core.IsSmartContractAddress(receiver) {
+	if core.IsSmartContractAddress(receiver) && isASCIIString(parsedESDTTransfers.CallFunction) {
 		responseParse.Function = parsedESDTTransfers.CallFunction
 	}
 
-	if len(parsedESDTTransfers.ESDTTransfers) == 0 {
+	if len(parsedESDTTransfers.ESDTTransfers) == 0 || !isASCIIString(string(parsedESDTTransfers.ESDTTransfers[0].ESDTTokenName)) {
```

Comment:
You could extract these checks in two functions which can be also used in `parsers/dataField/parseSingleNFTTransfer.go`

Reply:
cannot, because checks from SingleNFTTransfer are different